eng/common/scripts/get-codeowners.ps1

```diff
@@ -0,0 +1,43 @@
+param (
+  $TargetDirectory, # should be in relative form from root of repo. EG: sdk/servicebus
+  $RootDirectory # ideally $(Build.SourcesDirectory)
+)
+
+$codeOwnersLocation = @(Get-ChildItem -R -Force -Path $RootDirectory -Filter "CODEOWNERS")
+
+if ($codeOwnersLocation.Length -eq 0) {
+  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
+  exit 1
+}
+
+if ($codeOwnersLocation.Length -gt 1) {
+  Write-Host "Multiple CODEOWNERS files detected in  $RootDirectory."
+  Write-Host "$codeOwnersLocation"
+  exit 1
+}
+
+
+$codeOwnersContent = Get-Content ($codeOwnersLocation | Select-Object -Last 1)
+
+$ownedFolders = @{}
+
+foreach ($contentLine in $codeOwnersContent) {
+  if (-not $contentLine.StartsWith("#") -and $contentLine){
+    $splitLine = $contentLine -split "\s" | ? { return $_ }
+    
+    # in the codeowners file, gh aliases start with @. we don't want that when passing them to the API
+    $ownedFolders[$splitLine[0].ToLower()] = ($splitLine[1..$($splitLine.Length)] | % { return $_.substring(1) }) -join ","
```

Comment:
This logic seems very fragile. See https://github.com/Azure/azure-sdk-for-net/pull/13249 as an example of something that will end up breaking this logic.

I suggest you filter the list to things that start with `@` and then only take those.

Reply:
Great example, didn't know this was a thing. I believe you can't get around the fact that the first item HAS to be the id for the line though (the applicable path). Going to adjust to use your suggestion.